tests/CarbonSleeper/SleepTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Tests\Carbon;
+
+/*
+ * This file is part of the Carbon package.
+ *
+ * (c) Brian Nesbitt <brian@nesbot.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Carbon\Carbon;
+use Carbon\CarbonSleeper;
+use Tests\AbstractTestCase;
+
+class SleepTest extends AbstractTestCase
+{
+    protected function tearDown()
+    {
+        Carbon::setTestNow();
+        parent::tearDown();
+    }
+
+    public function testFakeSleepDelay()
+    {
+        Carbon::setTestNow(Carbon::now());
+
+        $before = time();
+
+        $carbonBefore = Carbon::now();
+
+        CarbonSleeper::sleep(5);
+
+        $after = time();
+
+        $carbonAfter = Carbon::now();
+
+        $this->assertLessThanOrEqual(
+            1, // Allow up to 1 second in case of timing issue between calls
+            $after - $before,
+            'Carbon sleep called system sleep in test condition'
+        );
+    }
+
+    public function testFakeSleepMoveTestTime()
+    {
+        Carbon::setTestNow(Carbon::now());
+
+        $before = Carbon::now()->getTimestamp();
+
+        CarbonSleeper::sleep(5);
+
+        $after = Carbon::now()->getTimestamp();
+
+        $this->assertEquals(
+            5,
+            $after - $before,
+            'Carbon sleep called system sleep in test condition'
+        );
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedMessage Number of seconds must be greater than or equal to 0
+     */
+    public function testNegativeSleep()
+    {
+        Carbon::setTestNow(Carbon::now());
+
+        CarbonSleeper::sleep(-1);
+    }
+
+    public function testRealSleep()
+    {
+        $before = time();
+
+        CarbonSleeper::sleep(1);
```

Comment:
Yes. The assertion message is for in case of failure, this is why we are expecting whatever we are asserting.

See: https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertEquals

> Reports an error identified by $message if the two variables $expected and $actual are not equal.

The message is not the success condition.  It is the failure condition.  The message is used when the test fails.

ie:
```php
    public function testRealSleep()
    {
        // Carbon::setTestNow(); - Wrapper left the test time set

        $before = time();

        CarbonSleeper::sleep(1);

        $after = time();

        $this->assertNotEquals($after, $before, 'Sleep did not delay execution');
    }
```

With message passed to the assertion
```sh
There was 1 failure:

1) Tests\CarbonSleeper\SleepTest::testRealSleep
Sleep did not delay execution
Failed asserting that 1447435126 is not equal to 1447435126.
```

Without message passed to the assertion
```sh
There was 1 failure:

1) Tests\CarbonSleeper\SleepTest::testRealSleep
Failed asserting that 1447435126 is not equal to 1447435126.
```

So, you are correct, the real sleep test does really call the real delay.  But that is not what the message is for.  It is for the case as documented in PHPUnit's documentation for when the message will be used.